app/main.go

```diff
@@ -27,22 +28,25 @@ func listSAs()([]LoadedIKE, error){
 	if err != nil {
 		return retVar, err
 	}
-	for _,m := range msgs.Messages() {
-		if e := m.Err(); e != nil{
+	for _, m := range msgs { // <- Directly iterate over msgs
+		if e := m.Err(); e != nil {
 			//ignoring this error
 			continue
 		}
 		for _, k := range m.Keys() {
 			inbound := m.Get(k).(*vici.Message)
 			var ike LoadedIKE
 			if e := vici.UnmarshalMessage(inbound, &ike); e != nil {
-				//ignoring this marshal/unmarshal errro!
+				//ignoring this marshal/unmarshal error!
 				continue
 			}
 			ike.Name = k
 			retVar = append(retVar, ike)
-
 		}
 	}
+
+	// close the socket
+	s.Close()
```

Comment:
This should not be used, as line 24 already registers the socket to be closed once the scope is left.

Reply:
Good point thanks. Ill update this to remove the close.